### PR TITLE
Deleted all sync clauses and await functions to avoid context problems.

### PR DESCRIPTION
**Pull request by Xelwon**
Deleted all sync clauses and await functions to avoid context problems.
NOTE: Collection, runtime and task are needed to can compile it in external ides as sharpdevelop.

### CODE REVIEW

File: Extensions/HttpClientExtensions.cs [hunk 1]
```diff
@@ -76,7 +78,7 @@ public static class HttpClientExtensions
             return parsedPayload;
         }
 
-        private static async Task<ResponseEnvelope> PerformRemoteProcedureCall<TRequest>(this System.Net.Http.HttpClient client,
+        public static  ResponseEnvelope PerformRemoteProcedureCall<TRequest>(this System.Net.Http.HttpClient client,
```
Comment (Lord-Haji) [inline Extensions/HttpClientExtensions.cs:81]: Why this extra space ?

File: Hash/LegacyHashser.cs [hunk 2]
```diff
@@ -7,11 +7,10 @@
 
 namespace PokemonGo.RocketAPI.Hash
 {
-    public class LegacyHashser   : IHasher
+    public class LegacyHashser : IHasher
     {
-        public async Task<HashResponseContent> RequestHashesAsync(HashRequestContent request)
+        public  HashResponseContent RequestHashes(HashRequestContent request)
```
Comment (Lord-Haji) [inline Hash/LegacyHashser.cs:12]: Here too

**Review by Lord-Haji — COMMENTED** [review]
I lots of extra spacies in Fort.cs too

File: Extensions/HttpClientExtensions.cs (hunk 1 above)
Comment (Xelwon) [reply Extensions/HttpClientExtensions.cs:81]: a mistake

File: Hash/LegacyHashser.cs (hunk 2 above)
Comment (Xelwon) [reply Hash/LegacyHashser.cs:12]: another mistake

Comment (Lord-Haji) [reply Hash/LegacyHashser.cs:12]: Resolve them?

**Comment by jjskuld** [comment]
This one will need more testing before we merge.

**Comment by Lord-Haji** [comment]
Yeah @jjskuld i was thinking of testing it.

**Comment by jjskuld** [comment]
This change does not make sense for a few reasons:
1) If you want synchronous behavior you can simply add `.Result` to each async API call. For example instead of `await Reauthenticate()` you call `Reauthenticate().Result`.
2) This would break all of the Rocket API clients today:  NecroBot.CLI, NecroBot.Win, RocketBot, etc.
3) It is the responsibility of your client to handle threading properly.  In the case of Winforms, you don't want to use `async MyAsyncTask()` but rather `Task.Run` to execute on another thread instead of GUI thread.
4) We are moving towards more async for performance.  Going back to synchronous is just going in wrong direction.

My feeling is that this is the wrong solution to fixing your issue.

**Comment by jjskuld** [comment]
Just wanted to get back to you regarding this issue. I've been reading up on this topic of how to properly design a library which does async work.

https://channel9.msdn.com/Events/TechEd/Europe/2013/DEV-B318

This video is very informative.  I believe the major issues that you are seeing are due to the lack of ConfigureAwait(false) in Rocket API.  So I will be fixing this in a separate pull request.  Thanks for your patience.

**Pull request closed without merging**